### ISSUE TITLE
lint: importShadow handling of renamed imports

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#304]
For packages that are imported with renaming, `importShadow` gives no warnings

If it's expected behavior, negative tests should be added.
If it should trigger on them, fix should be attached to this issue.

Reproducer:
```go
package checker_test

import (
	xmath "math"
)

var _ = xmath.E // Make sure xmath is used

//TODO: shadow of imported package 'xmath'
func renamedImportShadow(xmath int) {}
```
